x-pack/plugins/search_indices/public/hooks/api/use_indices_status.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useQuery } from '@tanstack/react-query';
+
+import { GET_STATUS_ROUTE } from '../../../common/routes';
+import type { IndicesStatusResponse } from '../../../common/types';
+
+import { useKibana } from '../use_kibana';
+
+const DEFAULT_INDICES_POLLING_INTERVAL = 15 * 1000;
+
+export const useIndicesStatusQuery = (pollingInterval = DEFAULT_INDICES_POLLING_INTERVAL) => {
+  const { http } = useKibana().services;
+  return useQuery({
+    refetchInterval: pollingInterval,
+    refetchIntervalInBackground: true,
```

Comment:
suggestion: use `refetchOnWindowFocus:'always'` so we fetch when window is active vs in background always polling